server/tests/modules/test_countbydate.py

```diff
@@ -10,25 +10,61 @@ def setUp(self):
         super(CountValuesTests, self).setUp()  # log in
 
         # test data designed to give different output if sorted by freq vs value
-        count_csv = 'Date,Amount,Foo\nJan 10 2016,10,Foo\nJul 25 2011,5,Goo\nJan 10 2016,1,Hoo\n'
+        count_csv = 'Date,Amount,Foo\nJan 10 2011,10,Foo\nJul 25 2016,5,Goo\nJan 10 2011 01:00:00,1,Hoo\nJan 10 2011 00:00:01,1,Hoo\nJan 10 2011 00:00:01,1,Hoo\nJan 10 2011 00:01:00,1,Hoo\nJan 15 2011,1,Too\n'
         workflow = create_testdata_workflow(count_csv)
 
         self.wf_module = load_and_add_module('countbydate', workflow=workflow)
         self.col_pval = get_param_by_id_name('column')
         self.sort_pval = get_param_by_id_name('sortby')
+        self.group_pval = get_param_by_id_name('groupby')
+        self.csv_data = get_param_by_id_name('csv')
 
     def test_render(self):
         # sort by value.
         # Use out.to_csv() instead of str(out) to ensure rows are output in index order (otherwise variable)
         set_string(self.col_pval, 'Date')
-        set_integer(self.sort_pval,0)  # 0 = sort by value
-        out = execute_wfmodule(self.wf_module)
-        self.assertEqual(out.to_csv(index=False), 'date,count\n2011-07-25,1\n2016-01-10,2\n' )
 
         # sort by freq
-        set_integer(self.sort_pval,1)  # 1 = sort by freq
+        set_integer(self.sort_pval, 1)  # 1 = sort by freq
+        out = execute_wfmodule(self.wf_module)
```

Comment:
No need to do this now, but we could and should make countbydate and external module. One step in doing that would be to call render directly, like https://github.com/CJWorkbench/filter/blob/master/test_filter.py#L23